### PR TITLE
feat: add getAccessKind

### LINKED ISSUE #397: 🚀 Feature: Add getAccessKind
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I took a peek through the [typescript-eslint `yarn.lock`](https://github.com/typescript-eslint/typescript-eslint/blob/b615fc295c05599085979e2064c6907645715e21/yarn.lock) to see who depends on `tsutils` still. [`eslint-plugin-deprecation`](https://github.com/gund/eslint-plugin-deprecation) does still for exactly one function: [`import { isReassignmentTarget } from 'tsutils';`](https://github.com/gund/eslint-plugin-deprecation/blob/5a32ce0c761782902c07ec9df4dee5150f7b5c4c/src/rules/deprecation.ts#L22).

[`isReassignmentTarget` is a one-liner in `tsutils`](https://github.com/ajafff/tsutils/blob/03b4df15d14702f9c7a128ac3fae77171778d613/util/util.ts#L1172-L1174) that calls to a [much larger `getAccessKind function`](https://github.com/ajafff/tsutils/blob/03b4df15d14702f9c7a128ac3fae77171778d613/util/util.ts#L1064).

### Additional Info

This is essentially a split-off from #3, long after I'd thought #3 was resolved. 😄

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #397
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Adds the function, with some rudimentary unit tests.

### CODE REVIEW

File: src/nodes/access.test.ts
Comment (JoshuaKGoldberg) [inline src/nodes/access.test.ts:1]: I'll add some more tests 🙂

**Comment by CHE1RON** [comment]
It would be great to have this (and through this, get `isReassignmentTarget`) for migrating https://github.com/cartant/eslint-plugin-etc 😉

File: src/nodes/access.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+import ts from "typescript";
+
+import { isAssignmentKind } from "../syntax";
+
+/* eslint-disable perfectionist/sort-enums */
+/**
+ * What operations(s), if any, an expression applies.
+ */
+export enum AccessKind {
```
Comment (RebeccaStevens) [inline src/nodes/access.ts:12]: Could it be better to express these in bitwise notation, to make it clear that's how they are supposed to be used?

```ts
export enum AccessKind {
  None = 0,
	Read = 1 << 0,
	Write = 1 << 1,
	Delete = 1 << 2,
	ReadWrite = Read | Write,
}
```

Comment (RebeccaStevens) [reply src/nodes/access.ts:12]: It's probably a bit unnecessary though.

**Review by RebeccaStevens — APPROVED** [review]

File: src/nodes/access.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/nodes/access.ts:12]: I like it!

**Pull request merged**